### ISSUE TITLE
ReferenceArrayInput replace of undefined (basepath)

### ISSUE THREAD

**Issue by gvillo** [marmelab/react-admin#3883]
**What you were expecting:**
ReferenceArrayInput is not working fine on v3 beta.3. It's failing on a line with a FIXME comment on `ra-core/esm/controller/input/useReferenceArrayInputController.js` file
```javascript
var referenceBasePath = basePath.replace(resource, reference); // FIXME obviously very weak
``` 

**What happened instead:**
It's not rendering at all.

**Steps to reproduce:**
Just add a filter including a ReferenceArrayInput to a List and run it.

**Related code:**

```html
  <Filter {...props}>
    <TextInput label="Search" source="q" alwaysOn />
    <ReferenceArrayInput label="Tags" source="tagsIds" reference="tags">
      <SelectArrayInput optionText="description" />
    </ReferenceArrayInput>
  </Filter>
```

**Environment**

* React-admin version: v3.0.0-beta.3
* Last version that did not exhibit the issue (if applicable):
* React version: v16.10.2
* Browser: Google Chrome 77.0.3865.120 (Official Build) (64-bit)
* Stack trace (in case of a JS error):

```javascript
Uncaught TypeError: Cannot read property 'replace' of undefined
    at useReferenceArrayInputController (useReferenceArrayInputController.js:136)
    at ReferenceArrayInput (ReferenceArrayInput.js:257)
    at renderWithHooks (react-dom.development.js:16367)
    at mountIndeterminateComponent (react-dom.development.js:18731)
    at beginWork$1 (react-dom.development.js:20061)
    at HTMLUnknownElement.callCallback (react-dom.development.js:363)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:412)
    at invokeGuardedCallback (react-dom.development.js:465)
    at beginWork$$1 (react-dom.development.js:25711)
    at performUnitOfWork (react-dom.development.js:24616)
    at workLoopSync (react-dom.development.js:24592)
    at performSyncWorkOnRoot (react-dom.development.js:24153)
    at react-dom.development.js:12313
    at unstable_runWithPriority (scheduler.development.js:818)
    at runWithPriority$2 (react-dom.development.js:12259)
    at flushSyncCallbackQueueImpl (react-dom.development.js:12308)
    at flushSyncCallbackQueue (react-dom.development.js:12296)
    at discreteUpdates$1 (react-dom.development.js:24334)
    at discreteUpdates (react-dom.development.js:1485)
    at dispatchDiscreteEvent (react-dom.development.js:5949)
useReferenceArrayInputController @ useReferenceArrayInputController.js:136
ReferenceArrayInput @ ReferenceArrayInput.js:257
renderWithHooks @ react-dom.development.js:16367
mountIndeterminateComponent @ react-dom.development.js:18731
beginWork$1 @ react-dom.development.js:20061
callCallback @ react-dom.development.js:363
invokeGuardedCallbackDev @ react-dom.development.js:412
invokeGuardedCallback @ react-dom.development.js:465
beginWork$$1 @ react-dom.development.js:25711
performUnitOfWork @ react-dom.development.js:24616
workLoopSync @ react-dom.development.js:24592
performSyncWorkOnRoot @ react-dom.development.js:24153
(anonymous) @ react-dom.development.js:12313
unstable_runWithPriority @ scheduler.development.js:818
runWithPriority$2 @ react-dom.development.js:12259
flushSyncCallbackQueueImpl @ react-dom.development.js:12308
flushSyncCallbackQueue @ react-dom.development.js:12296
discreteUpdates$1 @ react-dom.development.js:24334
discreteUpdates @ react-dom.development.js:1485
dispatchDiscreteEvent @ react-dom.development.js:5949
index.js:1375 The above error occurred in the <ReferenceArrayInput> component:
    in ReferenceArrayInput (created by Field)
    in Field (created by FilterFormInput)
    in div (created by FilterFormInput)
    in FilterFormInput (created by FilterForm)
    in form (created by FilterForm)
    in FilterForm (created by ReactFinalForm)
    in ReactFinalForm (created by EnhancedFilterForm)
    in EnhancedFilterForm (created by Filter)
    in Filter (at TemplatesList.js:18)
    in TemplatesFilter (at TemplatesList.js:29)
    in div (created by ForwardRef(Toolbar))
    in ForwardRef(Toolbar) (created by WithStyles(ForwardRef(Toolbar)))
    in WithStyles(ForwardRef(Toolbar)) (created by ListToolbar)
    in ListToolbar (created by ListView)
    in div (created by ListView)
    in ListView (created by List)
    in List (at TemplatesList.js:27)
    in TemplatesList (created by WithPermissions)
    in WithPermissions (created by Context.Consumer)
    in Route (created by ResourceRoutes)
    in Switch (created by ResourceRoutes)
    in ResourceRoutes (created by Resource)
    in Resource (at App.js:72)
    in Route (created by RoutesWithLayout)
    in Switch (created by RoutesWithLayout)
    in RoutesWithLayout (created by Context.Consumer)
    in div (created by Layout)
    in main (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in Layout (created by WithStyles(Layout))
    in WithStyles(Layout) (created by Context.Consumer)
    in withRouter(WithStyles(Layout)) (created by ConnectFunction)
    in ConnectFunction (created by LayoutWithTheme)
    in ThemeProvider (created by LayoutWithTheme)
    in LayoutWithTheme (at Layout.js:12)
    in Unknown (created by Context.Consumer)
    in Route (created by CoreAdminRouter)
    in Switch (created by CoreAdminRouter)
    in div (created by CoreAdminRouter)
    in CoreAdminRouter (created by Context.Consumer)
    in Route (created by CoreAdmin)
    in Switch (created by CoreAdmin)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by CoreAdmin)
    in TranslationProvider (created by CoreAdmin)
    in Provider (created by CoreAdmin)
    in CoreAdmin (at App.js:41)
    in App (at src/index.js:8)
```

**Comment by djhi** [comment]
Thanks for reporting this. If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means:

* CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple)
* A sample application via GitHub

**Comment by fzaninotto** [comment]
Confirmed and reproduced, thanks for the report.

**Comment by fzaninotto** [comment]
Fixed by #3898
